anti_patterns/deny-warnings.md

```diff
@@ -0,0 +1,101 @@
+# `#![deny(warnings)]`
+
+## Description
+
+A well-intentioned crate author wants to ensure their code builds without
+warnings. So they annotate their crate root with
+
+## Example
+
+```rust
+#![deny(warnings)]
+
+// All is well.
+```
+
+## Advantages
+
+It is short and will stop the build if anything is amiss.
+
+## Drawbacks
+
+By disallowing the compiler to build with warnings, a crate author opts out of
+Rust's famed stability. Sometimes new features or old misfeatures need a change
+in how things are done, thus lints are written that `warn` for a certain grace
+period before being turned to `deny`. Sometimes APIs get deprecated, so their
+use will emit a warning where before there was none.
+
+ALl this conspires to potentially break the build whenever something changes –
+not only for the original developer, but also for those who wish to use their
+code.
```

Comment:
This isn't true: Cargo passes `--cap-lints` to rustc, which disables lints while building dependencies. See rust-lang/rfcs#1193.